2021/ToIntegerOrInfinity.js

```diff
@@ -14,5 +14,7 @@ module.exports = function ToIntegerOrInfinity(value) {
 	var number = ToNumber(value);
 	if ($isNaN(number) || number === 0) { return 0; }
 	if (!$isFinite(number)) { return number; }
-	return $sign(number) * floor(abs(number));
+	var integer = floor(abs(number));
+	if (integer === 0) { return 0; }
+	return $sign(number) * integer;
```

Comment:
hmm, i'm confused, if `number` is -0, then `sign(number)` would be -0, and `-0 * -0` is `+0`, no?

Reply:
ahh, even tho `Math.sign(-0)` is `-0`, my `$sign` helper returns `-1` here. got it.

Reply:
Yep, you beat me to it